### PR TITLE
Config tweaks

### PR DESCRIPTION
**Pull request by djc**
Follow-up from #2470 that also tries to clean up a bunch of stuff.

### CODE REVIEW

File: bin/tests/integration/config_tests.rs [hunk 1]
```diff
@@ -289,7 +285,7 @@ define_test_config!(ipv4_only);
 define_test_config!(ipv6_only);
 define_test_config!(openssl_dnssec);
 define_test_config!(ring_dnssec);
-#[cfg(feature = "hickory-resolver")]
+#[cfg(feature = "resolver")]
```
Comment (bluejekyll) [inline bin/tests/integration/config_tests.rs:288]: I think we'll need to make sure this is listed as a breaking change, right?

**Review by bluejekyll — APPROVED** [review]
If we're going to move all of this into the binary as a new library, I'm wondering if we should review how we have all the utility binaries setup as well. That is, it would probably be nice to start creating common command line arguments, etc, across all the binaries in the hickory project. Do you have any thoughts on what we should do there?

Otherwise this looks like a good way to cleanup features.

File: bin/tests/integration/config_tests.rs (hunk 1 above)
Comment (bluejekyll) [reply bin/tests/integration/config_tests.rs:288]: oh, never mind, it was always `resolver` in the binary, this is fixing it to match as it's moved into the other crate.

**Comment by djc** [comment]
> If we're going to move all of this into the binary as a new library, I'm wondering if we should review how we have all the utility binaries setup as well. That is, it would probably be nice to start creating common command line arguments, etc, across all the binaries in the hickory project. Do you have any thoughts on what we should do there?

For me the main motivation here is that I wanted to move code from the -server crate that is not actually useful for the -server library API surface out of that crate and into the crate where it is used. From a quick look, the utility binaries are all pretty small so it feels like there isn't enough complexity that sharing the code for it would be worth it. It would make sense to make sure the command-line interfaces align across all of these but it's not very high on my priority list.

**Pull request merged**